spec/features/posts_index_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe 'post#index', type: :feature do
+  describe 'post page' do
+    before(:each) do
+      @user = User.create!(name: 'Amare23', photo: 'my-photo2', bio: 'Microverse student22')
+      @post = Post.create!(title: 'Integ test22', text: 'differs much in using matchers...', author_id: @user.id)
+      @comment = Comment.create!(text: 'I need help!', author: @user, post: @post)
+      Like.create!(author_id: @user.id, post_id: @post.id)
+
+      visit(user_posts_path(@user))
+    end
+
+    it " should display users' username" do
+      expect(page).to have_content('Amare')
+    end
+
+    it 'should display the number of posts for the user to be 1' do
+      expect(page).to have_content('Number of posts: 1')
+    end
+
+    it 'should not display the number of posts for the user to be 2' do
+      expect(page).to_not have_content('Number of posts: 2')
+    end
+
+    it "\nshould display posts author name" do
+      expect(page).to have_content(@post.author.name)
+    end
+
+    it "\nshould display posts title" do
+      expect(page).to have_content(@post.title)
+    end
+
+    it "\nshould display the body of the post text" do
+      expect(page).to have_content 'differs much in using'
+    end
+
+    it "\nshould display the comment on this post" do
+      expect(page).to have_content('I need help!')
+    end
+
+    it "\nshould display the comments counter for the post to be 1" do
+      expect(page).to have_content("Comments: #{@post.comments_counter}")
+    end
+
+    it "\nshould display the likes counter for the post to be 1" do
+      expect(page).to have_content('Likes: 1')
+    end
+
+    it "\nshould display post details when the user clicks title of the post link" do
```

Comment:
- [ ] You have 3 failing tests for features, kindly update them 
![image](https://user-images.githubusercontent.com/62705947/217491031-a0da3747-6621-41ad-b6c1-6b3b61743dce.png)

Reply:
- [ ] I know that this is outside of this milestone scope but out of all 77 tests you have 20 failing if we include controllers tests as well